lib/immutable-struct.rb

```diff
@@ -88,7 +88,12 @@ def self.new(*attributes,&block)
 
       define_method(:==) do |other|
         return false unless other.is_a?(klass)
-        attributes.all? { |attribute| self.send(attribute) == other.send(attribute) }
+        attributes.all? do |attribute|
+          if attribute.kind_of?(Array) and attribute.size == 1
```

Comment:
That's right... for array attributes, we're trying to `send` that to the instance `send([:widgets])`...